lib/extract/attribute_builder.rb

```diff
@@ -98,6 +99,11 @@ def string(value)
       raise ArgumentError, message if !value.is_a?(String) || value.length.zero?
     end
 
+    def in_range?(value, range)
+      message = "Invalid #{name}: must be a number between #{range.first} and #{range.last}"
```

Comment:
:star: Using `range.begin` and `range.end` would be a little better than `range.first` and `range.last`: `begin` and `end` are actually the properties you instantiated the range with. `first` and `last` both interact with the range as if it were an array. In fact, they would implicitly convert the range to an array and return the first and last values (❗️)  except that they have the special property that, if you pass no arguments to them, they act as aliases for `begin` and `end`.

To a reader of this code casually familiar with `Range`'s API, `begin` and `end` would be less surprising than `first` and `last`.